packages/app-headless-cms/src/admin/views/contentEntries/ContentEntry/FullScreenContentEntry/FullScreenContentEntryHeaderLeft.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useMemo } from "react";
+import { ReactComponent as BackIcon } from "@material-design-icons/svg/round/arrow_back.svg";
+import { useNavigateFolder } from "@webiny/app-aco";
+import { IconButton } from "@webiny/ui/Button";
+import { Typography } from "@webiny/ui/Typography";
+import { useModel } from "~/admin/components/ModelProvider";
+import { useContentEntry } from "~/admin/views/contentEntries/hooks";
+import {
+    EntryMeta,
+    EntryTitle,
+    EntryVersion,
+    TitleWrapper,
+    EntryName
+} from "./FullScreenContentEntry.styled";
+
+export const FullScreenContentEntryHeaderLeft = () => {
+    const { model } = useModel();
+    const { entry } = useContentEntry();
+    const { navigateToFolder, currentFolderId } = useNavigateFolder();
+
+    const title = useMemo(
+        () => entry?.meta?.title || `New ${model.name}`,
+        [entry.meta, model.name]
+    );
+    const isNewEntry = useMemo(() => !entry.meta?.title, [entry.meta]);
+    const version = useMemo(() => entry.meta?.version ?? null, [entry.meta]);
+    const status = useMemo(() => entry.meta?.status ?? null, [entry.meta]);
```

Comment:
All three hooks will always run, if `entry.meta` changes. You might as well combine all 3 into 1 hook, which returns an object `{ isNewEntry, version, status }`. Otherwise, use specific dependencies on each hook, like `entry.meta?.title`, `entry.meta?.version`, etc.